skfda/preprocessing/dim_reduction/_fdm.py

```diff
@@ -0,0 +1,229 @@
+"""Functional Diffusion Maps Module."""
+from __future__ import annotations
+
+from typing import Callable
+
+import numpy as np
+import scipy
+from sklearn.utils.extmath import svd_flip
+from typing_extensions import Self
+
+from ..._utils._sklearn_adapter import BaseEstimator, InductiveTransformerMixin
+from ...representation import FData
+from ...typing._numpy import NDArrayFloat
+
+KernelFunction = Callable[[FData, FData | None], NDArrayFloat]
+
+
+class FDM(
```

Comment:
I forgot to mention that, but FDM is not a well-known acronym, such as PCA. In fact, FDM seems to be used for "Fast difussion models" of "Fuzzy difussion models" in the literature.  I propose to use `DiffusionMap` as the name, instead, given that the "functional" part is redundant, being in a functional data package.